test_unstructured/file_utils/test_filetype.py

```diff
@@ -548,3 +550,260 @@ def magic_from_buffer_(request: FixtureRequest):
+        self, file_name: str | None
+    ):
+        with open(example_doc_path("ideas-page.html"), "rb") as f:
+            file = io.BytesIO(f.read())
+            if file_name is not None:
+                file.name = file_name
+
+        assert _FileTypeDetectionContext(file=file).extension == ""
+
+    # -- .file_head ---------------------------------------------
+
+    def it_grabs_the_first_4k_bytes_of_the_file_for_use_by_magic(self):
+        ctx = _FileTypeDetectionContext(file_path=example_doc_path("norwich-city.txt"))
+
+        head = ctx.file_head
+
+        assert isinstance(head, bytes)
+        assert len(head) == 4096
+        assert head.startswith(b"Iwan Roberts\nRoberts celebrating after")
+
+    # -- .file_path ---------------------------------------------
+
+    @pytest.mark.parametrize("file_path", [None, "a/b/c.pdf"])
+    def it_knows_the_file_path_provided_by_the_caller(self, file_path: str | None):
+        assert _FileTypeDetectionContext(file_path=file_path).file_path == file_path
+
+    # -- .mime_type ---------------------------------------------
+
+    def it_provides_the_MIME_type_detected_by_libmagic_from_a_file_path(self):
+        ctx = _FileTypeDetectionContext(file_path=example_doc_path("norwich-city.txt"))
+        assert ctx.mime_type == "text/plain"
+
+    def and_it_provides_the_MIME_type_from_path_using_filetype_lib_when_magic_is_unavailable(self):
+        with patch("unstructured.file_utils.filetype.LIBMAGIC_AVAILABLE", False):
+            ctx = _FileTypeDetectionContext(file_path=example_doc_path("simple.doc"))
+            assert ctx.mime_type == "application/msword"
+
+    def but_it_warns_to_install_libmagic_when_the_filetype_lib_cannot_detect_the_MIME_type(
+        self, caplog: LogCaptureFixture
+    ):
+        with patch("unstructured.file_utils.filetype.LIBMAGIC_AVAILABLE", False):
+            ctx = _FileTypeDetectionContext(file_path=example_doc_path("norwich-city.txt"))
+            assert ctx.mime_type is None
+            assert "WARNING" in caplog.text
+            assert "libmagic is unavailable" in caplog.text
+            assert "consider installing libmagic" in caplog.text
+
+    def it_provides_the_MIME_type_detected_by_libmagic_from_a_file_like_object(self):
+        with open(example_doc_path("norwich-city.txt"), "rb") as f:
+            ctx = _FileTypeDetectionContext(file=f)
+            assert ctx.mime_type == "text/plain"
+
+    def and_it_provides_the_MIME_type_from_file_using_filetype_lib_when_magic_is_unavailable(self):
+        with patch("unstructured.file_utils.filetype.LIBMAGIC_AVAILABLE", False):
+            file_path = example_doc_path("simple.doc")
+            with open(file_path, "rb") as f:
+                ctx = _FileTypeDetectionContext(file=f)
+                assert ctx.mime_type == "application/msword"
+
+    # -- .open() ------------------------------------------------
```

Comment:
Should there be any tests showing the file is closed appropriately?

Reply:
We only want it to be closed when it's from a file-path, otherwise we need to leave it open.

But I can put a line to do that in the `"..._when_it_is_a_file_path()"` case.